src/v/datalake/schema_parquet.cc

```diff
@@ -0,0 +1,179 @@
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i32_type{},
+          .logical_type = serde::parquet::date_type{},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::time_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i64_type{},
+          .logical_type = serde::parquet::
+            time_type{.is_adjusted_to_utc = false, .unit = serde::parquet::time_unit::micros},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::timestamp_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i64_type{},
+          .logical_type = serde::parquet::
+            time_type{.is_adjusted_to_utc = false, .unit = serde::parquet::time_unit::micros},
+        };
+    }
+
+    serde::parquet::schema_element
+    operator()(const iceberg::timestamptz_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i64_type{},
+          .logical_type = serde::parquet::
+            time_type{.is_adjusted_to_utc = true, .unit = serde::parquet::time_unit::micros},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::string_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::byte_array_type{},
+          .logical_type = serde::parquet::string_type{},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::uuid_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::byte_array_type{.fixed_length = 16},
+          .logical_type = serde::parquet::uuid_type{},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::fixed_type& type) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::byte_array_type{.fixed_length = type.length},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::binary_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::byte_array_type{},
+        };
+    }
+};
+
+serde::parquet::schema_element
+struct_to_parquet(const iceberg::struct_type& schema);
+serde::parquet::schema_element
+field_to_parquet(const iceberg::nested_field& field);
+
+struct type_converting_visitor {
+    serde::parquet::schema_element
+    operator()(const iceberg::primitive_type& type) {
+        auto res = std::visit(primitive_type_converting_visitor{}, type);
+        res.repetition_type
+          = required == required
+              ? serde::parquet::field_repetition_type::required
+              : serde::parquet::field_repetition_type::optional;
+        return res;
+    }
+
+    serde::parquet::schema_element
+    operator()(const iceberg::struct_type& type) {
+        return struct_to_parquet(type);
+    }
+
+    serde::parquet::schema_element operator()(const iceberg::list_type& type) {
```

Comment:
Lists are translated a little differently actually. There is a logical type for them and there is extra structure to get them to match: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists